### ISSUE TITLE
[Feature] Enable `Scene Tagger` to have its own default view

### ISSUE THREAD

**Issue by Dark-Obsidian** [stashapp/stash#4220]
**Is your feature request related to a problem? Please describe.**
- Current limitation is that as the `Scene Tagger` is a seen only as another view of the Scenes page (where `disp=3`)
- This means whatever is selected as the default view, applies to both **_Browsing scenes_** (in "Grid" view) and **_Tagging scenes_** (in the "Tagger" view)
- I would like to set the default view of the `Scene Tagger` to filter where `stash_id is null`, however doing this then filters all un-tagged scenes from the all other Scene views too

**Describe the solution you'd like**
- Keep it that saved filters are still shared across all Scene views (this is fine)
- But add a patch which lets stash set: [1] A default view for `Scene Tagger (disp=3)` and [2] A default view for `Other views (Grid|List|Wall)`

**Describe alternatives you've considered**
- _Could_ "hard-code" the `Scene Tagger` view to only show un-tagged scenes, however I wouldn't really suggest hard-coding anything as I realise that some users may not appreciate this

**Additional context**
- I tried writing a userscript that would force-inject the filter parameter into the url...
- So whenever the user browsed to a url like: `http://localhost:9999/scenes?<<params>>disp=3`
- They would be redirected to `http://localhost:9999/scenes?c=("type":"stash_id_endpoint","modifier":"IS_NULL")&<<params>>disp=3`
- However this did not seem to work and stash got stuck in a redirection loop

**Comment by Dark-Obsidian** [comment]
Hi @DingDongSoLong4 -- thanks for the reply...

***

### RE: Saved filters per `disp=`
Sure, get what you're saying and have done the same thing myself but the crux of the request was really around being able to set a different ***default*** saved filter for Scene Tagger if possible, as have found myself...
1. Going to `Scenes` (default Scene saved filter applies)
2. Opening `Scene Tagger`
3. Doing a `Scrape All` / `Search All`
4. Realising I have scraped/searched scene that are already tagged
5. Having to select my `Scene Tagger Default` saved filter
6. Re-run the `Scrape All` / `Search All`
7. Go through and tag my scenes

So the goal of this request is to essentially eliminate possibility of having to go through steps steps 4 - 6 and streamline the tagging process.

***

### RE: Redirection loop
Thanks for the suggestion, I thought I had covered this - see rule export below...
- Match: `http://localhost:*/scenes?*disp=3&z=0`
- Redirect: `http://localhost:$1/scenes?c=("type":"stash_id_endpoint","modifier":"IS_NULL")$2&disp=3&z=0`
- Exclude: `http://localhost:*c=*`

In the process of writing the above though, I actually spotted the error...
- Issue: I missed off an `&` after the `c=( ... )` insertion
- ❌ Redirect: `http://localhost:$1/scenes?c=("type":"stash_id_endpoint","modifier":"IS_NULL")$2&disp=3&z=0`
- ✅ Redirect: `http://localhost:$1/scenes?c=("type":"stash_id_endpoint","modifier":"IS_NULL")  &  $2&disp=3&z=0`

**Comment by DingDongSoLong4** [comment]
@Dark-Obsidian

This is my usual tagging "workflow":
1. Go to `Scenes` (applying the regular saved filter)
2. Click `Saved filters` > `Tagger Studio`.
This is a saved filter which sets `?c=("type":"studios","modifier":"IS_NULL")&sortby=created_at&sortdir=desc&disp=3`: filter by `Studios is null`, sort by `Created At (desc)`, and the Tagger view (`disp=3`). I also have a few other Tagger saved filters, each with different filters (eg `Stash ID is null`), but the rest is the same.
3. Tag scenes.

If I understand correctly, you want this:
1. Go to `Scenes`
2. Click the `Tagger` icon, and automatically have a tagger-specific default filter applied (ie for me, the `Tagger Studio` filter)
3. Tag scenes.

Yes my workflow has an extra click (the `Saved filters` button), but I don't really think that's a big deal.

From a user expectation perspective, the Tagger is a separate "view" in the scenes page, and since saved filters also include what "view" you're in, it doesn't in my opinion make sense to have separate default filters for only the Tagger.

Perhaps the solution is to completely separate the Tagger from the other views, where then it would make sense to have separate default filters. You wouldn't use the `Tagger` view to browse, and you wouldn't use the `Grid`/`List`/`Wall` views to tag scenes - there's a clear logical distinction there, but they are all treated "equally" as views.

**Comment by Dark-Obsidian** [comment]
Thanks @DingDongSoLong4, some good points to consider...

***

### Your approach: 👍🏼
- Thanks for sharing info on your workflow
- I appreciate the way you get to `Scene Tagger` from a saved filter (rather than by selecting the `Scene Tagger` icon directly and may try to adapt that way of working!

***

### "From a user expectation perspective, the Tagger is just another "view" in the scenes page"
- Ah, so that's a very interesting point as I actually see things differently (no personal, just IMO)...
- I ***DON'T*** see `Scene Tagger` as *just another view of the scenes page* as IMO, there are specific tasks that can only be done from this page
- It is especially useful to highlight `Scene Tagger` for new joiners for example, given the advantages of going to this page rather than going to `Scene > Scrape With` etc
- (FYI: This is part of the reason I have specifically highlighted the `Scene Tagger` button in my own stash)...
![Scene tagged highlighted using custom CSS](https://user-images.githubusercontent.com/140384474/275783516-d7277854-e837-427e-96fc-872f468cf30b.png)

### User "mode"
I also think that it is worth considering the user's "mode", i.e. primary thing they are trying to achieve at this moment in time...

#### 1. Browse/View
- The user wants to watch porn
- They use the `Grid`, `List`, `Wall` views to navigate to porn they want to watch
- Scene Tagger: ❌ They are very unlikely to go into `Scene Tagger` when in this 'mode'

#### 2. Organise/Import
- The user wants to organise their porn, complete or update some metadata, etc
- Scene Tagger: ✅ They go to the `Scene Tagger` specifically to tag scenes and populate metadata

**Comment by DingDongSoLong4** [comment]
I agree with you entirely - the use cases for `Grid`, `List` and `Wall` are entirely separate from the Tagger. The Tagger serves a very different purpose, that being organising/importing content, rather than viewing it.

What I meant was that this distinction isn't being clearly communicated in the UI - `Grid`, `List`, `Wall` and `Tagger` are all grouped together as views of the `Scenes` page. Your highlighting of the `Scene Tagger` button (and the fact that we have to tell people "click here to efficiently tag your scenes") I think shows a deficiency in the UI design/layout.

I think the Tagger should be clearly separated from the other "views" - maybe it could have an entirely different layout (eg using a different header), a different way of accessing it - something to clearly show that it is for a separate "user mode" as you put it. I'm not a UI designer by any means - these are just some ideas that I imagine could improve the overall experience.

**Comment by scruffynerf** [comment]
+1 to @Dark-Obsidian 's take.  Let's do the Scene Tagger icon change asap, and then discuss bigger revamps
